### ISSUE TITLE
Do not remove custom directives from the model

### ISSUE THREAD

**Issue by lastmjs** [aerogear/graphback#1089]
It looks like custom directives that I place into my model are removed in the generated schema. The definitions are not removed, which is nice, but not very useful without keeping the directives themselves. It would be great if custom directives placed in the model could be kept in the generated schema. This was a major pain point with Prisma and implementing permissions with custom directives. They were removed, causing me to have to write a parser to place them back in so that I could have my directives executed against the generated schema, which I imagine is desirable for a variety of use cases.

**Comment by wtrocki** [comment]
Critical issue. Going to be resolved in the next release

**Comment by craicoverflow** [comment]
We are using `printSchema` to convert the schema object to a string before writing to a file. It [seems to be expected behaviour](https://github.com/graphql/graphql-js/issues/869#issuecomment-309899380) that custom directives are not added.

We should use something other than `printSchema` for this.

**Comment by wtrocki** [comment]
GraphQL-Toolkit has printSchema with directives. 
https://gist.github.com/arjun-kava/0bd38e0c8feec360284677b1ee31728a

Generally if we were using GraphQL-Compose for that purpose (as everywhere else) we will be fine

**Comment by craicoverflow** [comment]
As I said, GraphQL-Toolkit changes order completely and adds a `query` root type.

I tested with GraphQL-Compose and it has the same issue.

**Comment by craicoverflow** [comment]
Yeah I already tested these - GraphQL-Toolkit changes print order and adds a root `query` type. GraphQL-Compose also loses directives.

**Comment by craicoverflow** [comment]
Directives are getting removed from fields in SchemaComposer. When we add the schema string or object to the constructor as we do now (see example below) then the final schema will have directive definitions, but no directives in use.

```ts
import { SchemaComposer } from 'graphql-compose'
import { loadSchemaSync } from '@graphql-toolkit/core';

const schema = loadSchemaSync(`directive @test(
  reason: String = "No longer supported"
) on FIELD_DEFINITION | ENUM_VALUE

type ModifyMe {
  id: ID! @test(reason: "asjdk")
}

type Query {
  hello: ModifyMe
}`, { loaders: [] })

const schemaComposer = new SchemaComposer(schema);

const schemaStr = schemaComposer.toSDL({ exclude: ['ID', 'String', 'Int', 'Boolean', 'Float'] })

console.log(schemaStr)
```

This can be resolved by _adding_ typedefs string separately:

```ts
const schemaString = `directive @test(
  reason: String = "No longer supported"
) on FIELD_DEFINITION | ENUM_VALUE

type ModifyMe {
  id: ID! @test(reason: "asjdk")
}

type Query {
  hello: ModifyMe
}`;

const schemaComposer = new SchemaComposer();

schemaComposer.addTypeDefs(schemaString)
```

However this method only take a string argument whereas we are operating on a GraphQLSchema object.

The best approach here is operate on a string schema always in Graphback. 

For code generation this can be a very minimal change in the CLI, by changing `loadSchema(model)` to `loadSchema(model, 'string')`:
 https://github.com/aerogear/graphback/blob/2756b29adb3696a99b499574299fc9b2f35e0d36/packages/graphback-cli/src/components/generate.ts#L74

For runtime this would be a breaking change, since the runtime API accepts string or object, however loading the model from the template should go away in https://github.com/aerogear/graphback/issues/1071

**Comment by wtrocki** [comment]
This sounds like good bug to log for graphql-compose. Operating on SDL strings in plugin executor mean that every plugin will need to build schema from SDL which is not ideal.

**Comment by craicoverflow** [comment]
I've logged https://github.com/graphql-compose/graphql-compose/issues/246

**Comment by nodkz** [comment]
Should be fixed in graphql-compose@7.14.2
